src/libraries/illink-sharedframework.targets

```diff
@@ -30,8 +30,10 @@
       <ILLinkArgs>$(ILLinkArgs) --nowarn $(LinkerNoWarn)</ILLinkArgs>
     </PropertyGroup>
 
-     <!-- Retrieve CoreLib's targetpath via GetTargetPath as it isn't binplaced yet. -->
+    <!-- Retrieve CoreLib's path as it isn't binplaced alongside the libraries. -->
     <MSBuild Projects="$(CoreLibProject)"
+             Properties="Configuration=$(RuntimeConfiguration)"
```

Comment:
With this, the Configuration property is always passed in as a global property even if the RuntimeConfiguration already matches the current (libraries) configuration. We should keep this in sync with how we set global properties for the ProjectReference to CoreLib.

Reply:
```
         <Properties Condition="'$(RuntimeFlavor)' == 'CoreCLR' and 
                                '$(Configuration)' != '$(CoreCLRConfiguration)'">Configuration=$(CoreCLRConfiguration)</Properties> 
         <Properties Condition="'$(RuntimeFlavor)' == 'Mono' and 
                                '$(Configuration)' != '$(MonoConfiguration)'">Configuration=$(MonoConfiguration)</Properties> 
```